Variant & Experiment searches

laceysanderson opened the pull request:
This PR adds the following searches:
 - Experiment/Project: `research/projects`
 - Variant: `search/variants`

## Preparation
To test these searches you will first need to remove the previous views.
1. Go to `admin/structure/views` and search for each view
2. Click on the arrow beside edit and choose disable

## Testing
- The search should appear at the same url as the old view -make sure the link on the frontpage takes you to the new search (after prep is done)
- All filter criteria should work (note for experiments the genus filter is dependant on a property of taxrank:genus. I've added these to portal but they may not be on your clone; species is disabled on experiment search)

## Screenshots
<img width="1028" alt="Screen Shot 2020-04-02 at 7 58 30 PM" src="https://user-images.githubusercontent.com/1566301/78316316-935b7500-751c-11ea-8afe-81bb32bcb9ca.png">
<img width="1013" alt="Screen Shot 2020-04-02 at 8 00 08 PM" src="https://user-images.githubusercontent.com/1566301/78316323-95253880-751c-11ea-83ea-d2d34d7d862b.png">

reynoldtan commented:
Preps in my local KP Clone:
1. Right modules
![Screen Shot 2020-04-17 at 11 25 55 AM (2)](https://user-images.githubusercontent.com/15472253/79597312-468ca800-809f-11ea-97d9-c98ad9206112.png)
2. Disabled views
![Screen Shot 2020-04-17 at 11 29 08 AM](https://user-images.githubusercontent.com/15472253/79597340-57d5b480-809f-11ea-93fb-c7dc9310ecd8.png)
3. In the branch
![Screen Shot 2020-04-17 at 11 29 38 AM](https://user-images.githubusercontent.com/15472253/79597359-5f955900-809f-11ea-9266-32e4ded457d5.png)
4. Cleared cache several times, also re-installed the modules then repeat the preps.
Tried in fresh too, but variants redirect to search/node.

Result page:
search/variants
![Screen Shot 2020-04-17 at 11 25 15 AM (2)](https://user-images.githubusercontent.com/15472253/79597389-6d4ade80-809f-11ea-8e49-8a71636fd817.png)

research/projects
![Screen Shot 2020-04-17 at 10 55 27 AM (2)](https://user-images.githubusercontent.com/15472253/79597478-8f446100-809f-11ea-8fd6-3b76031e3cb7.png)

Comments:
1. Crop thumbnails point to a directory in kp_frontpage, maybe add kp_frontpage as dependencies.

laceysanderson commented:
Hmmm... I'm not sure why it's not working on your local clone. I've set it up on fresh based on the same instructions here and it works 👍 Do you want to just complete your review there?

reynoldtan commented:
Tested in /fresh and components work as described. 

**In Research/Projects**
No issues encountered when viewing the page with enough horizontal spacing. It is when you reduce the screen size that layout seems to break.

Summary text of each experiment gets cut off and some texts and digit from the metadata to the right, gets pushed to a new line unnecessarily. In some cases when text (especially in the tag icon) becomes long, it overlaps with entries below it.

![Screen Shot 2020-04-27 at 7 33 26 PM (2)](https://user-images.githubusercontent.com/15472253/80437128-0ce94780-88be-11ea-811d-9713260bfaaf.png)

Suggestion:
1. Add helpful text (title attribute) to icons to provide user textual meaning.
2. Drop select species field since it stayed disabled all the time for any crops.
3. A search title same as in Search Variants.

**In Search/Variants**
Works! but I have no clue about fields under Genome Range since it has no example like the rest of the form fields.

Suggestions:
1. A default crop similar to research/project (lentil).
2. A new line to the text "While...." since it extends beyond the form alignment.
3. Reset or clear form from previous search when switching crops.

laceysanderson commented:
Thank you for the review @reynoldtan! I have addressed all your suggestions except removing the disabled species drop-down which I'm keeping for consistency with other searches. Could you give it a quick check on dev/fresh?

reynoldtan commented:
Beautiful! one last tiny issue, in experiment with less summary tends to collapse the metadata icons.

![Screen Shot 2020-05-07 at 7 44 49 AM (2)](https://user-images.githubusercontent.com/15472253/81301997-03d34580-9037-11ea-86be-ce220113bd76.png)

laceysanderson commented:
Fixed! Thanks you for the thorough review!
<img width="1149" alt="Screen Shot 2020-05-07 at 10 52 21 AM" src="https://user-images.githubusercontent.com/1566301/81322225-e27f5300-9050-11ea-9b11-c1cce0fa5bfc.png">
